tools/ports/sqlite3.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+import os
+import shutil
+import logging
+
+# sqlite amalgamation download URL uses relase year and tag
+# 2022  and (3, 38, 5) -> '/2022/sqlite-amalgamation-3380500.zip'
+VERSION = (3, 38, 5)
+VERSION_YEAR = 2022
+HASH = '4fc2992e4c1ca1664a9b01e07c9b944003c4ed0612978e471eff262a7114e4a0699244d91e71ae4ad2b5e1fc9829917cd7d5f0313aa5859036905f974548d94a'
+
+deps = []
+
+
+def needed(settings):
+  return settings.USE_SQLITE3
+
+
+def get_lib_name(settings):
+  return 'libsqlite3' + ('-mt' if settings.USE_PTHREADS else '') + '.a'
+
+
+def get(ports, settings, shared):
+  release = f'sqlite-amalgamation-{VERSION[0]}{VERSION[1]:02}{VERSION[2]:02}00'
+  # TODO: Fetch the file from an emscripten-hosted mirror.
+  ports.fetch_project('sqlite3', f'https://www.sqlite.org/{VERSION_YEAR}/{release}.zip', release, sha512hash=HASH)
+
+  def create(final):
+    logging.info('building port: libsqlite3')
+
+    source_path = os.path.join(ports.get_dir(), 'sqlite3', release)
+    dest_path = os.path.join(ports.get_build_dir(), 'sqlite3')
+
+    shutil.rmtree(dest_path, ignore_errors=True)
+    shutil.copytree(source_path, dest_path)
+
+    ports.install_headers(dest_path)
+
+    # flags are based on sqlite-autoconf output.
+    # SQLITE_HAVE_ZLIB is only used by shell.c
+    flags = [
+      '-DSTDC_HEADERS=1',
+      '-DHAVE_SYS_TYPES_H=1',
+      '-DHAVE_SYS_STAT_H=1',
+      '-DHAVE_STDLIB_H=1',
+      '-DHAVE_STRING_H=1',
+      '-DHAVE_MEMORY_H=1',
+      '-DHAVE_STRINGS_H=1',
+      '-DHAVE_INTTYPES_H=1',
+      '-DHAVE_STDINT_H=1',
+      '-DHAVE_UNISTD_H=1',
+      '-DHAVE_FDATASYNC=1',
+      '-DHAVE_USLEEP=1',
+      '-DHAVE_LOCALTIME_R=1',
+      '-DHAVE_GMTIME_R=1',
+      '-DHAVE_DECL_STRERROR_R=1',
+      '-DHAVE_STRERROR_R=1',
+      '-DHAVE_POSIX_FALLOCATE=1',
+      '-DSQLITE_OMIT_LOAD_EXTENSION=1',
+      '-DSQLITE_ENABLE_MATH_FUNCTIONS=1',
+      '-DSQLITE_ENABLE_FTS4=1',
+      '-DSQLITE_ENABLE_FTS5=1',
+      '-DSQLITE_ENABLE_RTREE=1',
+      '-DSQLITE_ENABLE_GEOPOLY=1',
+      '-DSQLITE_OMIT_POPEN=1',
+    ]
+    if settings.USE_PTHREADS:
+      flags += [
+        '-sUSE_PTHREADS=1',
+        '-DSQLITE_THREADSAFE=1',
+        '-D_REENTRANT=1',
```

Comment:
I don't think you need to add `-D_REENTRANT=1` explicitly since that is set automatically in `-pthreads` builds.

Reply:
I have removed the line.